src/triangle.rs

```diff
@@ -0,0 +1,356 @@
+            triangle: *self,
+            ix: 0,
+        }
+    }
+
+    #[inline]
+    fn area(&self) -> f64 {
+        Triangle::area(self)
+    }
+
+    #[inline]
+    fn perimeter(&self, _accuracy: f64) -> f64 {
+        self.a.distance(self.b) + self.b.distance(self.c) + self.c.distance(self.a)
+    }
+
+    #[inline]
+    fn winding(&self, pt: Point) -> i32 {
+        let s0 = (self.b - self.a).cross(pt - self.a).signum();
+        let s1 = (self.c - self.b).cross(pt - self.b).signum();
+        let s2 = (self.a - self.c).cross(pt - self.c).signum();
+
+        if s0 == s1 && s1 == s2 {
+            s0 as i32
+        } else {
+            0
+        }
+    }
+
+    #[inline]
+    fn bounding_box(&self) -> Rect {
+        Rect::new(
+            self.a.x.min(self.b.x.min(self.c.x)),
+            self.a.y.min(self.b.y.min(self.c.y)),
+            self.a.x.max(self.b.x.max(self.c.x)),
+            self.a.y.max(self.b.y.max(self.c.y)),
+        )
+    }
+
+    #[inline]
+    fn as_rect(&self) -> Option<Rect> {
+        Some(self.bounding_box())
+    }
+}
+
+// NOTE: vertices a, b and c are not garunteed to be in order as described in the struct comments
+//       (i.e. as "vertex a is topmost, vertex b is leftmost, and vertex c is rightmost")
+impl Iterator for TrianglePathIter {
+    type Item = PathEl;
+
+    fn next(&mut self) -> Option<PathEl> {
+        self.ix += 1;
+        match self.ix {
+            1 => Some(PathEl::MoveTo(self.triangle.a)),
+            2 => Some(PathEl::LineTo(self.triangle.b)),
+            3 => Some(PathEl::LineTo(self.triangle.c)),
+            4 => Some(PathEl::ClosePath),
+            _ => None,
+        }
+    }
+}
+
+// TODO: better and more tests
+#[cfg(test)]
+mod tests {
+    use crate::{Point, Triangle, Vec2};
+
+    fn assert_approx_eq(x: f64, y: f64) {
+        assert!((x - y).abs() < 1e-7);
+    }
+
+    #[test]
+    fn centroid() {
+        let test = Triangle::from_coords((-90.02, 3.5), (7.2, -9.3), (8.0, 9.1)).centroid();
+        let expected = Point::new(-24.939999999999998, 1.0999999999999996);
+
+        assert_eq!(test, expected);
+    }
+
+    #[test]
+    fn offsets() {
+        let test = Triangle::from_coords((-20.0, 180.2), (1.2, 0.0), (290.0, 100.0)).offsets();
+        let expected = [
+            Vec2::new(-110.39999999999999, 86.8),
+            Vec2::new(-89.19999999999999, -93.39999999999999),
+            Vec2::new(199.60000000000002, 6.6000000000000085),
+        ];
+
+        assert_eq!(test, expected);
+    }
+
+    #[test]
+    fn area() {
+        let test = Triangle::new(
+            (12123.423, 2382.7834),
+            (7892.729, 238.459),
+            (7820.2, 712.23),
+        )
+        .area();
+        let expected = 1079952.91574081;
+
+        assert_approx_eq(test, expected);
+    }
+
+    #[test]
+    fn circumcenter() {
+        let test = Triangle::EQUILATERAL.circumcenter();
+        let expected = Point::new(0.5, 0.2886751345948128);
```

Comment:
This should be approx equal, otherwise the test is fragile. I see it's fine for incircle.